
User space that includes the ext2 defines (e.g. syslinux) must bot get
ext2_fs_sb since it contains in-kernel locking structures ...

  - Rene Rebe <rene@exactcode.de>

--- usr/include/linux/ext2_fs.h.orig	2005-01-07 09:21:12.471622368 +0100
+++ usr/include/linux/ext2_fs.h	2005-01-07 09:21:31.036800032 +0100
@@ -17,7 +17,9 @@
 #define _LINUX_EXT2_FS_H
 
 #include <linux/types.h>
+#ifdef __KERNEL__
 #include <linux/ext2_fs_sb.h>
+#endif
 
 /*
  * The second extended filesystem constants/structures
